qemu/target-i386/translate.c

```diff
@@ -4768,16 +4767,16 @@ static target_ulong disas_insn(CPUX86State *env, DisasContext *s,
     }
 
     // Unicorn: trace this instruction on request
-    if (env->uc->hook_insn) {
-        trace = hook_find(env->uc, UC_HOOK_CODE, pc_start);
-        if (trace) {
-            if (s->last_cc_op != s->cc_op) {
-                sync_eflags(s, tcg_ctx);
-                s->last_cc_op = s->cc_op;
-                changed_cc_op = true;
-            }
+    if (HOOK_EXISTS(env->uc, UC_HOOK_CODE)) {
```

Comment:
yeah, this is actually a bug. HOOK_EXISTS should also have a bounds check variant.